File: Arrays/Python/MoveAllZerosToEnd.py
def MoveAllZerosToEnd(nums,n):
    j=0
    for i in range(n):
        if(nums[i]!=0):
            nums[i],nums[j]=nums[j],nums[i]
            j +=1
    return nums


n=int(input())
nums=list(map(int,input().split()))
print(MoveAllZerosToEnd(nums,n))

File: Arrays/C++/MoveAllZerosToEnd.cpp
#include<bits/stdc++.h>

using namespace std;

void MoveAllZerosToEnd(vector<int> &arr, int n){
    int i,j=0;
        
    for(i=0;i<n;i++){
        if(arr[i]!=0){
            swap(arr[i],arr[j]);
            j++;
        }
    }
    
}

int main(){
    
    int i,n,m,res;
    cin>>n;
    vector<int> nums;
    for(i=0;i<n;i++){
        cin>>m;
        nums.push_back(m);
    }
    
    MoveAllZerosToEnd(nums,n);
    
    for(i=0;i<n;i++){
        cout<<nums[i]<<endl;
    }
    
    
    
    
}